backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/VdsManager.java

```diff
@@ -1331,7 +1334,19 @@ public void updateV2VJobInfos(List<V2VJobInfo> v2vJobInfos) {
     }
 
 
-    private void initAvailableCpus() {
+    private void updateCpuTopology() {
+
+        if (!cpuTopology.isEmpty()) {
```

Comment:
in the DB i think it can be null, therefore we may pass null from cachedVds somehow? i prefer being safe and also check null here, although it does initiate above.

Reply:
I don't think it is possible - there is a check below that checks if the `cpuTopology` from the `cachedVds` is `null`. The `cpuTopology` not being `null` is a very hard constraint - It would fail on every call to `getCpuTopology()` when the topology is copied (as I've discovered while creating the patch :-D  )

Reply:
right right.. we do have the check below :)